libethereum/ClientTest.cpp

```diff
@@ -101,8 +102,17 @@ void ClientTest::modifyTimestamp(int64_t _timestamp)
 
 void ClientTest::mineBlocks(unsigned _count)
 {
+    std::promise<void> allBlocksImported;
+    int blocksLeftToImport = _count;
+    auto importHandler =
+        setOnBlockImport([&blocksLeftToImport, &allBlocksImported](BlockHeader const&) {
+            if (--blocksLeftToImport == 0)
+                allBlocksImported.set_value();
+        });
     m_blocksToMine = _count;
     startSealing();
+    allBlocksImported.get_future().wait_for(
+        std::chrono::seconds(m_singleBlockMaxMiningTimeInSeconds * _count));
```

Comment:
We should return the error (from `ClientTest::mineBlocks` and from RPC method) in case timeout expired.
See `future::wait_for` return value.